hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/bucket/OMBucketCreateRequest.java

```diff
@@ -314,7 +314,9 @@ private BucketEncryptionInfoProto getBeinfo(
    */
   private void checkQuotaInNamespace(OmVolumeArgs omVolumeArgs,
       long allocatedNamespace) throws IOException {
-    if (omVolumeArgs.getQuotaInNamespace() != OzoneConsts.QUOTA_RESET) {
+    // Client has ensured that no negative values other than -1 (clear quota)
+    // will occur when quota is set.
+    if (omVolumeArgs.getQuotaInNamespace() > OzoneConsts.QUOTA_RESET) {
```

Comment:
Why we change above condition check? Above check just means that we enabled the quota, and current quota isn't equal to default value. This should be a more accurate check.

Reply:
@linyiqun Here with omVolumeArgs.getQuotaInNamespace () > 0 is enough? Because when it > 0 is certainly is not equal to -1.

Reply:
@captainzmc , here my point is that we should firstly check if the quota was enabled by using one variable flag(here I understand is OzoneConsts.QUOTA_RESET). Then we check the quota value if it's valid. Is there a switch config for the quota now?  If the quota is not enabled, we don't need to do the quota check anymore.

Reply:
Current there are two case  -1 or -2 (old volume/bucket) not check quota, actually we have guarantee on the client, [user cann’t set negative quota](https://github.com/apache/ozone/blob/master/hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/client/OzoneQuota.java#L194), Therefore, except for clear quota and Old Quoat, there is no negative case.

Reply:
> Current there are two case -1 or -2 (old volume/bucket) not check quota, actually we have guarantee on the client, [user cann’t set negative quota](https://github.com/apache/ozone/blob/master/hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/client/OzoneQuota.java#L194), Therefore, except for clear quota and Old Quoat, there is no negative case.
> 
>If add omVolumeArgs. getQuotaInNamespace ()! = 2. The judgment condition will increase too much and the code will not be clear enough, we can use omVolumeArgs. getQuotaInNamespace () > 0.

Reply:
Okay, go ahead for this.